src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -587,9 +576,11 @@ public static void Reverse<T>(ref T elements, nuint length)
             ReverseInner(ref elements, length);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static void ReverseInner<T>(ref T elements, nuint length)
         {
-            Debug.Assert(length > 0);
+            if (length <= 1)
```

Comment:
This can be `if (length == 0) { return }` since zero is the only other value it could be here. It will also be a more efficient check.

Reply:
I added comparison with 1 since reversing one element is a nop, but it probably doesn't really make a difference.